### PR TITLE
feat: add webpack template generator

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
feat

**Did you add tests for your changes?**
yet to

**If relevant, did you update the documentation?**
yet to 

**Summary**
Adds a handler to scaffold a webpack repo from the cli, basically invokes webpack-defaults and generates stuff, I'm not sure if we need everything so open for feedback.

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: packages/generators/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "plugin-template"
   ],
   "dependencies": {
+    "@webpack-contrib/defaults": "^6.3.0",
```
Comment (alexander-akait) [inline packages/generators/package.json:25]: No need do this, we should reimplement logic and template here

Comment (anshumanv) [reply packages/generators/package.json:25]: gotcha wip

**Comment by alexander-akait** [comment]
I think we should implement this step by step, let's start with simple template with default files like `.editorconfig`/`package.json`/`ling-staged`/`husky`/`LICENSE`/`jest.config.js`/`prettierrc.js`/`eslintrc.js`/etc

**Comment by alexander-akait** [comment]
Copy here, webpack-default will be deprecated in favor `webpack loader --template webpack` and ``webpack plugin --template webpack`

**Comment by rishabh3112** [comment]
> Copy here, webpack-default will be deprecated in favor webpack loader --template webpack and webpack plugin --template webpack

We should replace the `default` templates for `loader` and `plugin` only imo, as current implementation is dependant on `webpack-defaults` only and we are copying for there only.

**Comment by anshumanv** [comment]
I'll remove dependency on `webpack-defaults` plus we don't need everything from there, will get to this in some time

File: packages/generators/init-template/webpack-template/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "webpack-template",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {},
```
Comment (jamesgeorge007) [inline packages/generators/init-template/webpack-template/package.json:5]: Add a `prepare` script to automatically enable Git hooks.
```suggestion
  "scripts": {
    "prepare": "husky install"
  },
```

File: packages/generators/init-template/webpack-template/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "@commitlint/cli": "^12.1.4",
+    "@commitlint/config-conventional": "^12.1.4",
+    "eslint": "^7.26.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-prettier": "^3.4.0",
+    "jest": "^26.6.3",
+    "prettier": "^2.3.0",
+    "webpack": "^5.37.1",
+    "webpack-cli": "^4.7.0"
+  },
```
Comment (jamesgeorge007) [inline packages/generators/init-template/webpack-template/package.json:16]: Add entries for `husky` and `lint-staged`.

File: packages/generators/src/handlers/webpack-defaults.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// These files will be directly copied to the destination dir
+const files = [
+    ".husky",
+    ".editorconfig",
+    ".eslintrc.js",
+    "prettier.config.js",
+    "jest.config.js",
+    "LICENSE",
+    "lint-staged.config.js",
+];
```
Comment (jamesgeorge007) [inline packages/generators/src/handlers/webpack-defaults.ts:24]: Add `.gitignore` to the list.

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
Left a couple of suggestions.

File: packages/generators/init-template/webpack-template/lint-staged.config.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  '*.js': ['eslint --fix', 'prettier --write', 'git add'],
+  '*.{json,md,yml,css,ts}': ['git add', 'prettier --write'],
```
[the remark this answers is not in the document]
Comment (anshumanv) [reply packages/generators/init-template/webpack-template/lint-staged.config.js:3]: Addressed all changes here - https://github.com/webpack/webpack-cli/pull/2854

**Comment by anshumanv** [comment]
Stale, closing in favour of https://github.com/webpack/webpack-cli/pull/2854

**Pull request closed without merging**